node/include/cocaine/middleware/auth.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cocaine/context.hpp>
+#include <cocaine/hpack/header.hpp>
+
+#include "cocaine/api/auth.hpp"
+
+namespace cocaine {
+namespace middleware {
+
+class auth_t {
+    std::shared_ptr<api::auth_t> auth;
+    std::string service;
+
+public:
+    explicit
+    auth_t(context_t& context, std::string service) :
+        auth(api::auth(context, "core")),
+        service(std::move(service))
+    {}
+
+    template<typename Event, typename F, typename... Args>
+    auto
+    operator()(F& fn, Event, const std::vector<hpack::header_t>& meta, Args&&... args) ->
```

Comment:
Still don't get why it is a passed variable instead of simple template parameter.
middleware<Event>(...); seems to be quite straightforward